pkg/etcd/etcd.go

```diff
@@ -853,6 +887,161 @@ func (e *ETCD) Snapshot(ctx context.Context, config *config.Control) error {
 		}
 	}
 
+	snapshots, err := e.listSnapshots(ctx, snapshotDir)
+	if err != nil {
+		return err
+	}
+
+	// store the snapshot data somewhere
+	_ = snapshots
+
+	return nil
+}
+
+// snapshotFile represents a single snapshot and it's
+// metadata.
+type snapshotFile struct {
+	Name      string `json:"name"`
+	NodeName  string `json:"nodeName"`
+	CreatedAt string `json:"createdAt"`
```

Comment:
Kubernetes has a metav1.Time that is handles serializing/deserializing timestamps to/from json and protobuf:
https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#Time

Reply:
Thanks.